src/Partials/AlignBoxButton.vala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2018 Alecaddd (http://alecaddd.com)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * Authored by: Giacomo "giacomoalbe" Alberini <giacomoalbe@gmail.com>
+ */
+public class Akira.Partials.AlignBoxButton : Gtk.Grid {
+  public string icon_name;
```

Comment:
you can also make it so that when this property is set, the actual icon is updated using the vala property syntax. You can check an example here:

https://github.com/Philip-Scott/Spice-up/blob/09a734117957c6f79fbf251dc349a5916fffb45a/src/Widgets/SlideWidget.vala#L25

https://wiki.gnome.org/Projects/Vala/PropertiesSample


This will prevent the need for the `change_icon_style` function down bellow